HealthMonitoring.SelfHost/Controllers/EndpointsController.cs

```diff
@@ -82,14 +82,23 @@ public IEnumerable<EndpointDetails> GetEndpoints()
         }
 
         [Route("api/endpoints/{id}/tags")]
-        [ResponseType(typeof (Guid))]
-        [SwaggerResponse(HttpStatusCode.Created, Type = typeof (Guid))]
+        [SwaggerResponse(HttpStatusCode.OK, Type = typeof (EndpointDetails))]
+        [SwaggerResponse(HttpStatusCode.NotFound)]
         [SwaggerResponse(HttpStatusCode.BadRequest)]
-        public IHttpActionResult PostUpdateEndpointTags(Guid id, [FromBody]TagsModel model)
+        [ResponseType(typeof(EndpointDetails))]
+        public IHttpActionResult PostUpdateEndpointTags(Guid id, [FromBody]string[] tags)
         {
-            model.ValidateModel();
-            _endpointRegistry.UpdateEndpointTags(id, model.Tags);
-            return Ok(EndpointDetails.FromDomain(_endpointRegistry.GetById(id)));
+            tags.CheckForUnallowedSymbols();
+            try
+            {
+                if(_endpointRegistry.TryUpdateEndpointTags(id, tags))
+                    return Ok(EndpointDetails.FromDomain(_endpointRegistry.GetById(id)));
+                return NotFound();
+            }
+            catch (Exception e)

```

Comment:
what exceptions are we expecting here?
if it's expected it should be a particular type eg. mycustomexception, or if its unexpected should be 5xx not 4xx?